src/Schema/Comparator.php

```diff
@@ -363,7 +363,29 @@ public function compareTables(Table $fromTable, Table $toTable): TableDiff
             );
         }
 
-        $renamedColumns = $this->detectRenamedColumns($addedColumns, $droppedColumns);
+        $renamedColumnNames = $toTable->getRenamedColumns();
+
+        foreach ($addedColumns as $addedColumnName => $addedColumn) {
+            if (! isset($renamedColumnNames[$addedColumn->getName()])) {
+                continue;
+            }
+
+            $removedColumnName = $renamedColumnNames[$addedColumn->getName()];
+            // Explicitly renamed columns need to be diffed, because their types can also have changed
+            $modifiedColumns[$removedColumnName] = new ColumnDiff(
+                $removedColumnName,
+                $addedColumn,
+                $this->diffColumn($droppedColumns[$removedColumnName], $addedColumn),
+                $droppedColumns[$removedColumnName],
+            );
+
+            unset(
+                $addedColumns[$addedColumnName],
+                $droppedColumns[$removedColumnName],
+            );
+        }
+
+        $this->detectRenamedColumns($modifiedColumns, $addedColumns, $droppedColumns);
```

Comment:
@stof 
> The automatic detection of renames should be kept IMO, as not all Schema diffing happens for a Schema that has been manipulating based on the introspected one.

It has been kept, it's just the internal handling of where the renamed columns are stored that is changed, see here